components/automate-ui/src/app/components/create-object-modal/create-object-modal.component.ts

```diff
@@ -52,37 +64,35 @@ export class CreateObjectModalComponent implements OnInit, OnDestroy, OnChanges
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    // update project dropdown if list changes
-    if (changes.assignableProjects) {
-      this.projects = {};
-      changes.assignableProjects.currentValue.forEach((proj: Project) =>
-        this.projects[proj.id] = { ...proj, checked: false });
-    }
-    // clear checked projects when opening
     if (changes.visible && (changes.visible.currentValue as boolean)) {
-      Object.values(this.projects).forEach(p => p.checked = false);
+
+      this.store.dispatch(new GetPolicies()); // refresh in case of updates
+
+      Object.values(this.projects).forEach(p => p.checked = false); // reset projects
       this.projectsUpdatedEvent.emit();
+
+      Object.values(this.policies).forEach(p => p.checked = false); // reset policies
+      this.policiesUpdatedEvent.emit();
+
       if (this.createProjectModal) {
-        this.updatePolicyCheckbox(true); // always set to checked upon opening
+        this.updatePolicyCheckbox(true); // default to checked upon opening
       }
     }
   }
 
-  onProjectChecked(project: ProjectChecked): void {
-    this.projects[project.id].checked = project.checked;
-    const projectsSelected = Object.keys(this.projects).filter(id => this.projects[id].checked);
+  onProjectDropdownClosing(projectsSelected: string[]): void {
     this.createForm.controls.projects.setValue(projectsSelected);
```

Comment:
Moved micro-managing into the dropdown where it belongs. We only have to set the value once, when the dropdown closes.